packages/@sanity/portable-text-editor/src/editor/plugins/createWithPortableTextMarkModel.ts

```diff
@@ -202,14 +202,31 @@ export function createWithPortableTextMarkModel(
           editor.onChange()
         }
       }
-      // Check consistency of markDefs
+      // Check consistency of markDefs (unless we are merging two nodes)
       if (
-        isTextBlock &&
-        editor.operations.some((op) =>
-          ['split_node', 'remove_node', 'remove_text', 'merge_node'].includes(op.type),
+        editor.isTextBlock(node) &&
+        !editor.operations.some(
+          (op) => op.type === 'merge_node' && 'markDefs' in op.properties && op.path.length === 1,
         )
       ) {
-        normalizeMarkDefs(editor, types)
```

Comment:
This is very pretty inefficient as it will do the whole editor, and not the actual nodes that are transformed.